# Binomial Distribution

## Dependency

In [7]:
import math

## Problem

We flip a coin 10 times, and got heads 2 times, and tails 8 times. Is this coin a fair coint?

Coin toss follows a binomial distribution. We can compute the probability of how many heads we get by the following formula. $k$ is the number of heads, $n$ is the number of total tosses, $p$ is the probability of getting a head.

$$
P(k) = \binom{n}{k} p^k (1 - p)^{n - k}
$$

where

$$
\binom{n}{k} = \frac{n!}{k! (n - k)!}
$$

When a coin is fair, $p = 0.5$. We got 2 heads out of 10 times, so $n = 10$ and $k = 2$. The formula gives us,

$$
\binom{10}{2} 0.5^2 (1 - 0.5)^8
$$

$\binom{10}{2}$ is,

$$
\binom{10}{2} = 
$$
$$
= \frac{10!}{2!8!}
$$
$$
= \frac{10 \times 9}{2}
$$
$$
= 45
$$

$0.5^2 = 0.25$ and $0.5^8$ is,

In [1]:
0.5**8

0.00390625

$\binom{10}{2} 0.5^2 (1 - 0.5)^8$ is

$$
= 45 \times 0.25 \times 0.0039
$$

In [3]:
45 * 0.25 * 0.5**8

0.0439453125

It's about 4%. If we think about an unfair coin like a head probability is 0.4, 

In [19]:
p = 0.4
n = 10
k = 2
ans = math.comb(n, k) * p**k * (1 - p)**(n - k)
print(f'{ans:.2%}')

12.09%


It's about 12%. With an unfair guy, we have more probability to get 2 heads out of 10 times. So it's likely that our coin is not fair.

## Resource

- [Checking whether a coin is fair](https://en.wikipedia.org/wiki/Checking_whether_a_coin_is_fair)